backend/src/apiserver/server/visualization_server.go

```diff
@@ -115,13 +112,14 @@ func (s *VisualizationServer) getVisualizationServiceURL(request *go_client.Crea
 	return s.serviceURL
 }
 
-func isVisualizationServiceAlive(serviceURL string) bool {
+func isVisualizationServiceAlive(serviceURL string) (bool, error) {
 	resp, err := http.Get(serviceURL)
-	if err != nil {
-		glog.Error("Unable to verify visualization service is alive!", err)
-		return false
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		return false, errors.New(fmt.Sprintf("Unable to verify visualization service is alive by sending request to %s and get %s !", serviceURL, resp.Status))
+	} else if resp == nil && err != nil {
```

Comment:
nit: `resp` is not necessarily nil when there is an `err`, but `resp` will not be nil when there isn't an `err`.

I'd suggest
```
if err != nil {
  // handle err
} else if resp.StatusCode != http.StatusOK {
  // we can assume resp is not nil now, because that's idiomatic go
  // e.g. you can read http.Get documentation it will mention this. https://golang.org/pkg/net/http/#Client.Get
}

return nil
```
```

Reply:
Good point, new revision of PR addressed the comment.